bisqapps/shared/presentation/src/commonMain/kotlin/network/bisq/mobile/presentation/ui/uicases/offers/CurrencyListPresenter.kt

```diff
@@ -0,0 +1,49 @@
+package network.bisq.mobile.presentation.ui.uicases.offers
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import network.bisq.mobile.domain.data.BackgroundDispatcher
+import network.bisq.mobile.domain.data.model.FiatCurrency
+import network.bisq.mobile.domain.data.repository.CurrenciesRepository
+import network.bisq.mobile.presentation.BasePresenter
+import network.bisq.mobile.presentation.MainPresenter
+import network.bisq.mobile.presentation.ui.navigation.Routes
+
+class CurrencyListPresenter(
+    mainPresenter: MainPresenter,
+    private val currenciesRepository: CurrenciesRepository,
+) : BasePresenter(mainPresenter), ICurrencyList {
+
+    private val _currencies = MutableStateFlow<List<FiatCurrency>>(emptyList())
+    override val currencies: StateFlow<List<FiatCurrency>> = _currencies
+
+    override fun onSelectedCurrency(currency: FiatCurrency) {
+        rootNavigator.navigate(Routes.OfferList.name)
+    }
+
+    private fun refresh() {
+        CoroutineScope(BackgroundDispatcher).launch {
+            try {
+                delay(1000) // TODO: To simulate loading. Yet to be handled
```

Comment:
General rule in mobile is don't show anything that the user cannot interact with. If there are no offers it should be hidden, but as you said let's leave this for fine tuning! ;)

Reply:
I was wrong its not 150 markets as we filter for those where we have price, i guess about 30- 40 then... added the filter in my new branch. But still agree we might find a better UX to not show empty markets in a fine tune phase.

Reply:
So there is filter option in the CurrencyListing (yet to be implemented).

In Bisq2 desktop, by default it shows all currencies.
But in mobile, we can set the `showmarkets` filter to `only with offers` by default.

<img width="369" alt="Screenshot 2024-11-26 at 10 28 28 AM" src="https://github.com/user-attachments/assets/cacd93d7-3721-4610-a9e0-7077d2d3f675">

and right now, I will hardcode all 150+ currnecies that are shown in Bisq2 desktop app.

Reply:
For loader, I was thinking of 2 scenarios.

Scenario 1:
1. A listing page, where all data needs to be coming from an api call.
2. In this case, we can just pass `isLoading = true` and `loadingWidget = {Component}` to The BisqScaffold/BisqLayout classes and they take care of rendering the given component in the mid of the screen till the data arrives.

Scenario 2:
1. A form page, where all controls can be rendered initially with empty fields,
2. An overlay loader can be shown with blurred background so the form/controls are visible in the background, but cannot be interacted with
3.  Once the data arrives, overlay loader is removed and the form is populated with actual data.

Even to achieve this, we set only `isLoading = true` to The BisqScaffold/BisqLayout classes. Without setting `loadingWidget` when isLoading is set to true, they show the overlay loader.

Is this understandable / sounds good? Please let me know.

Reply:
I would suggest to leave the loading topic for later. For the markets / currency list I doubt it will be needed (at node its instant, at client we can hard code data as its static data).
Of other screens like offer list, the list gets automatically updated with data, so data stream in. That is also the case in Desktop where at first start it takes a bit until the offers arrive and then fill up the list as they arrive.

But I think different user cases will have diff. context here, which will be easier to analyze once the core functionality is implemented.

Reply:
> So there is filter option

I would keep that also for later, it at all (IMO we should try to keep mobile to the most minimal feature set as possible).

> I will hardcode all 150+ currnecies that are shown in Bisq2 desktop app.

No need. I have it already connected with real data.

Reply:
okay

Reply:
@nostrbuddha your comment on the scenarios can go straight into the description of a new "Loading Spinners" UI issue ;)

Reply:
Created at https://github.com/bisq-network/bisq-mobile/issues/80